lib/core/include/scipp/core/value_and_variance.h

```diff
@@ -179,15 +179,24 @@ constexpr auto operator-(const T2 a, const ValueAndVariance<T1> b) noexcept {
 }
 template <class T1, class T2>
 constexpr auto operator*(const ValueAndVariance<T1> a, const T2 b) noexcept {
+  // Order of operations to prevent integer overflow of `b*b`.
   return ValueAndVariance{a.value * b, a.variance * b * b};
 }
 template <class T1, class T2>
 constexpr auto operator*(const T1 a, const ValueAndVariance<T2> b) noexcept {
-  return ValueAndVariance{a * b.value, a * a * b.variance};
+  // Order of operations to prevent integer overflow of `a*a`.
+  return ValueAndVariance{a * b.value, b.variance * a * a};
 }
 template <class T1, class T2>
 constexpr auto operator/(const ValueAndVariance<T1> a, const T2 b) noexcept {
-  return ValueAndVariance{a.value / b, a.variance / (b * b)};
```

Comment:
Would it be possible instead to use a trick like the order of operations above?
e.g. `return ValueAndVariance{a.value / b, a.variance / b / b};`?

Reply:
It would be. But this would trade a multiplication for a division. So it would be slightly more expensive. But I don't know if that matters. Should I change it?

Reply:
> Should I change it?

I don't mind, your call. Tradeoff between performance and code simplicity.

Reply:
The other code needs a comment, so it's not necessarily simpler :-)